src/hub.js

```diff
@@ -429,55 +429,56 @@ async function handleHubChannelJoined(entryManager, hubChannel, messageDispatch,
     });
 
     while (!scene.components["networked-scene"] || !scene.components["networked-scene"].data) await nextTick();
-    scene.components["networked-scene"]
-      .connect()
-      .then(() => {
-        let newHostPollInterval = null;
-
-        scene.emit("didConnectToNetworkedScene");
-        // When reconnecting, update the server URL if necessary
-        NAF.connection.adapter.setReconnectionListeners(
-          () => {
-            if (newHostPollInterval) return;
-
-            newHostPollInterval = setInterval(async () => {
-              const currentServerURL = NAF.connection.adapter.serverUrl;
-              const newHubHost = await hubChannel.getHost();
-              const newServerURL = `wss://${newHubHost}`;
-
-              if (currentServerURL !== newServerURL) {
-                console.log("Connecting to new Janus server " + newServerURL);
-                scene.setAttribute("networked-scene", { serverURL: newServerURL });
-                NAF.connection.adapter.serverUrl = newServerURL;
-              }
-            }, 1000);
-          },
-          () => {
-            clearInterval(newHostPollInterval);
-            newHostPollInterval = null;
-          },
-          null
-        );
-
-        NAF.connection.adapter.reliableTransport = (clientId, dataType, data) => {
-          const payload = { dataType, data };
-
-          if (clientId) {
-            payload.clientId = clientId;
-          }
 
-          hubChannel.channel.push("naf", payload);
-        };
-      })
-      .catch(connectError => {
-        // hacky until we get return codes
-        const isFull = connectError.error && connectError.error.msg.match(/\bfull\b/i);
-        console.error(connectError);
-        remountUI({ roomUnavailableReason: isFull ? "full" : "connect_error" });
-        entryManager.exitScene();
+    NAF.connection.onConnect(() => {
```

Comment:
Actually, I think it may be better to do this on the `"adapter-ready"` event I added recently, since this onConnect callback is essentially just `document.body.addEventListener("connected", ...)`.

Reply:
oh, you're right I guess this just puts it at the head of the list. adapter-ready seems smarter!